libs/seiscomp/processing/secondarypicker/S_v.cpp

```diff
@@ -0,0 +1,108 @@
+ * Public License version 3.0 as published by the Free Software Foundation *
+ * and appearing in the file LICENSE included in the packaging of this     *
+ * file. Please review the following information to ensure the GNU Affero  *
+ * Public License version 3.0 requirements will be met:                    *
+ * https://www.gnu.org/licenses/agpl-3.0.html.                             *
+ *                                                                         *
+ * Other Usage                                                             *
+ * Alternatively, this file may be used in accordance with the terms and   *
+ * conditions contained in a signed written agreement between you and      *
+ * gempa GmbH.                                                             *
+ ***************************************************************************/
+
+
+#define SEISCOMP_COMPONENT VPicker
+
+#include <seiscomp/logging/log.h>
+#include <seiscomp/processing/operator/ncomps.h>
+#include <seiscomp/math/filter.h>
+
+#include "S_v.h"
+
+
+using namespace std;
+
+namespace Seiscomp {
+
+namespace Processing {
+
+REGISTER_SECONDARYPICKPROCESSOR(SVPicker, "S-V");
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+SVPicker::SVPicker() : SAICPicker("V-AIC", Vertical) {}
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+SVPicker::~SVPicker() { }
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+bool SVPicker::setup(const Settings &settings) {
+	if ( !SecondaryPicker::setup(settings) ) return false;
+
+	try { setNoiseStart(settings.getDouble("spicker.V.noiseBegin")); }
+	catch ( ... ) {}
+
+	try { setSignalStart(settings.getDouble("spicker.V.signalBegin")); }
+	catch ( ... ) {}
+
+	try { setSignalEnd(settings.getDouble("spicker.V.signalEnd")); }
+	catch ( ... ) {}
+
+	AICConfig cfg = aicConfig();
+
+	try { cfg.threshold = settings.getDouble("spicker.V.threshold"); }
+	catch ( ... ) {}
+
+	try { cfg.minSNR = settings.getDouble("spicker.V.minSNR"); }
+	catch ( ... ) {}
+
+	try { cfg.margin = settings.getDouble("spicker.V.marginAIC"); }
+	catch ( ... ) {}
+
+	try { cfg.timeCorr = settings.getDouble("spicker.V.timeCorr"); }
+	catch ( ... ) {}
+
+	try { cfg.filter = settings.getString("spicker.V.filter"); }
+	catch ( ... ) { cfg.filter = "BW(4,0.3,1.0)"; }
+
+	try { cfg.detecFilter = settings.getString("spicker.V.detecFilter"); }
+	catch ( ... ) { cfg.detecFilter = "STALTA(1,10)"; }
+
+	return setAicConfig(cfg);
+}
+// <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+
+
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+WaveformOperator* SVPicker::createFilterOperator(Filter* compFilter)
```

Comment:
While thinking about it, this picker does not need an operator because it is just one component. `WaveformProcessor::setFilter` should be enough. We can keep it like that for the time being but it feels a bit overdesigned and introduces code that is actually not required.

Reply:
I agree and I thought thoroughly about it while creating the common base class for S_l2 and S_v. However this is the only way to make the common class work on both scenarios: it can handle multiple components but it is over designed for the basic scenario of a single one. I am happy to improve the code if you have some ideas.

Reply:
Actually `setFilter` would need to receive a `ChainFilter` then. Of course, this could be implemented in case `createFilterOperator` returns `nullptr`. But I agree that this would not make the code more readable. We can keep it as it is now.

Reply:
IMO the issue is that S_l2 wants to filter the data before performing the L2-norm and only after it applies the detector filter. I wondert if that is really required. Why not applying the frequency filter and the detector both on the L2-norm?

Reply:
I understand the result would be different, but I am questioning if that would be worse or not.

Reply:
Because the L2 norm is not symmetric, it is just positive. Furthermore we get rid of the offsets of each component when doing the filtering prior to combining.